AppDB/cassandra_utils.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+#
+# Simple script to rebalance the load on a Cassandra pool. This works only for
+# AppScale deployment.
+#
+# Author: graziano
+
+# Cassandra nodetool.
+CMD="/root/appscale/AppDB/cassandra/cassandra/bin/nodetool"
+
+# The keyspace to work on.
+KEYSPACE="Keyspace1"
+
+set -e
+
+help() {
+        echo "$0 {-cleanup|-repair}"
+        echo
+        echo "Utility for an AppScale Cassandra cluster:"
+        echo "   -repair               repair the cluster (should run routinely)."
+        echo "   -cleanup              cleanup the cluster (WARNING: it may take a long time)."
+        echo "   -help                 print this message."
+        echo
+}
+
+# This script can only run on the login node.
+am_i_login_node() {
+        if [ -e /etc/appscale/login_private_ip ]; then
+                LOGIN_IP="$(cat /etc/appscale/login_private_ip)"
+                for x in $(ip addr show|awk '/inet / {print $2}'|sed 's;/.*;;') ; do
+                        [ "$LOGIN_IP" = "$x" ] && return 0
+                done
+        fi
+        return 1
+}
+
+# Check if we are on the head node.
+if ! am_i_login_node ; then
+        echo "This script should run on the head node."
+        exit 1
+fi
+
+# Check for other running instances of this script.
```

Comment:
We should work with PID or lockfile instead of grepping and psing blindly. One option could be to save our PID and then remove it. If the PIDfile is there, we assume the process is running, and after too much time we can check if the PID is running.

The other option would be to use something lockfile to create a file and use it as a lock. Create file, make sure we obtain the lock on it, and then delete at the end of repair. If I cannot get the lock within few retries, quit.